source/scale/high-availability-cluster.rst

```diff
@@ -289,40 +289,69 @@ The database can be configured for high availability and transparent failover us
 Recommended configuration settings for PostgreSQL
 ``````````````````````````````````````````````````
 
-If you're using PostgreSQL as the choice of database, we recommend the following configuration optimizations on your Mattermost server. These  configurations were tested on an AWS Aurora r5.xlarge instance of PostgreSQL 11.7. There are also some general optimizations mentioned which requires servers with higher specifications.
+If you're using PostgreSQL as the choice of database, we recommend the following configuration optimizations on your Mattermost server. These configurations were tested on an AWS Aurora r5.xlarge instance of PostgreSQL 11.7. There are also some general optimizations mentioned which requires servers with higher specifications.
 
-1. **max_connections**: If you are using read-replicas, set reader connections to 1024, and set writer connections to 256. If you are using a single instance, then only setting reader connections to 1024 should be sufficient. If the instance is lower capacity than r5.xlarge, then set it to a lower number. Also tune the ``MaxOpenConns`` setting under the ``SqlSettings`` of the Mattermost app accordingly.
+**Config for Postgres Primary or Writer node**
 
-2. **random_page_cost**: Set it to 1.1, unless the DB is using spinning disks.
+.. code-block:: bash
 
-3. **work_mem**: Set it to 16MB for readers, and 32MB for writers. If it's a single instance, 16MB should be sufficient. If the instance is of a lower capacity than r5.xlarge, then set it to a lower number.
+  # If the instance is lower capacity than r5.xlarge, then set it to a lower number. 
+  # Also tune the "MaxOpenConns" setting under the "SqlSettings" of the Mattermost app accordingly. 
+  # Note that "MaxOpenConns" on Mattermost is per connection string.
```

Comment:
"per connection" may be confusing. Perhaps better to say "per database"?

Reply:
What i'm trying to say is "per connection string" but you're right it might be unclear. The setting it's truly per database if the customer is using RDS and chooses their endpoint, which is one string for N number of databases. 

@cwarnermm - any suggestions on making it more clear? I'm probably over thinking it in the name of clarity

TLDR; Taking the below config for example. There are 4 databases configured (`primary`, `replica-1`, `replica-2`, and `search-replica`). That means MaxOpenConns can do 300 * 4

```sql
      "SqlSettings": {
        "DataSource": "primary",
        "DataSourceReplicas": [
          "replica-1",
          "replica-2"
        ],
        "DataSourceSearchReplicas": [
          "search-replica"
        ],
        "MaxOpenConns": 300,
    },
```

But if we modify it to be the below, you have three strings with `replicas` representing `replica-1` and `replica-2`, it's just been abstracted away from Mattermost. So the max connections Mattermost will open here is 300 * 3. 

```sql
      "SqlSettings": {
        "DataSource": "primary",
        "DataSourceReplicas": [
          "replicas"
        ],
        "DataSourceSearchReplicas": [
          "search-replica"
        ],
        "MaxOpenConns": 300,
    },
```

Reply:
Maybe "per datasource endpoint"?

Reply:
"Per database endpoint" is precise and doesn't assume the number of databases in play. Out of curiosity, why is "per connection string" unclear?

Reply:
Per connection is confusing. Per connection string is technically fine, but the word generally used is dsn or datasource name or database endpoint. 0/5 on what to use.